### PR TITLE
Filter inherent transaction and not deposit excutive result event

### PR DESCRIPTION
**Pull request by gguoss**
https://github.com/paritytech/substrate/issues/1985

### CODE REVIEW

File: srml/executive/src/lib.rs [hunk 1]
```diff
@@ -184,7 +186,9 @@ impl<
 		// decode parameters and dispatch
 		let (f, s) = xt.deconstruct();
 		let r = f.dispatch(s.into());
-		<system::Module<System>>::note_applied_extrinsic(&r, encoded_len as u32);
+		if !is_inherent {
+			<system::Module<System>>::note_applied_extrinsic(&r, encoded_len as u32);
```
Comment (tomusdrw) [inline srml/executive/src/lib.rs]: This breaks multiple assumptions about the code, the function is also:
1. Increasing the current `extrinsic_index` in storage
2. Calculates total length of extrinsics.

If we really don't want to have the event it should rather be passed as a flag.

**Comment by bkchr** [comment]
Why should an inherent extrinsic not fail?

**Comment by gguoss** [comment]
> Why should an inherent extrinsic not fail?

Inherent fail == spawn block fail.
For example:  time set inherent can't fail.

Why do the failed inherent transactions need to be packaged into blocks?

**Comment by gavofyork** [comment]
there's no fundamental way of guaranteeing that an inherent won't fail. it's unlikely, the block-level protocol treats all extrinsics the same, in this way inherents are just as fallible as transactions.

**Pull request closed without merging**